sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -155,6 +157,12 @@ export class LinkEntity {
    * @returns {void}
    */
   protected async _ensureTokenRenewal(): Promise<void> {
+    // Clear the existing token renewal timer in case it hasn't already fired.
+    // This scenario can happen if the connection goes down and is brought back up
+    // before the `nextRenewalTimeout` was reached.
+    if (this._tokenRenewalTimer) {
+      clearTimeout(this._tokenRenewalTimer);
+    }
```

Comment:
I don't see this in Event Hubs. Don't we need this there as well?

Reply:
I believe the reason we didn't need this in event-hubs was because we don't have the onDetached logic there that would trigger replacing an existing timeout.

Reply:
But it doesnt harm us in adding it right? To future proof? It makes sense to clear the timer before trying to reset it. If you agree, then I recommend updating #10051 to include the change to Event Hubs as well.

Reply:
Right, it won't hurt us. Sure, I'll update #10051.